query/statement_rewriter.go

```diff
@@ -202,6 +202,30 @@ func rewriteShowSeriesCardinalityStatement(stmt *influxql.ShowSeriesCardinalityS
 		}
 	}
 
+	// do hll estimation for per-measurement queries
+	if !stmt.Exact && stmt.Dimensions == nil && stmt.Limit == 0 && stmt.Offset == 0 {
```

Comment:
I could take these conditions off (except exact) but I think it would muddy the waters.

`count_hll` works for any kind of query, but the optimized series iteration only works for no dimension/limit/offset.